components/automate-ui/src/app/pages/project/rules/project-rules.component.ts

```diff
@@ -234,19 +229,17 @@ export class ProjectRulesComponent implements OnInit, OnDestroy {
   }
 
   convertToRule(): Rule {
-    const conditions: Condition[] = [];
     // This constant ensures type safety
     const equals_op: ConditionOperator = 'EQUALS';
-    this.ruleForm.controls.conditions.value.forEach(c => {
-      conditions.push(<Condition>{
+    const conditions: Condition[] = this.ruleForm.controls.conditions.value.map(
+      (c: {attribute: string, values: string, operator: string}) => ({
         attribute: c.attribute,
         operator: c.operator,
           // Convert values string to storage format
         values: c.operator === equals_op
```

Comment:
Not great, but maybe slightly better than what we have now...?
```suggestion
        values: <ConditionOperator>c.operator === 'EQUALS'
```